cmake/FindFLAC.cmake

```diff
@@ -0,0 +1,22 @@
+# - Try to find FLAC
+# Once done this will define
+#
+# FLAC_FOUND - system has libFLAC
+# FLAC_INCLUDE_DIRS - the libFLAC include directory
+# FLAC_LIBRARIES - The libFLAC libraries
+
+find_package(PkgConfig)
```

Comment:
I spent a bit more time learning the ins and outs of cmake. In the main CMakeLists I think all we really need for each of those dependencies instead of pkg_search_module() is:
```
#once:
include(FindPackageHandleStandardArgs)
```
```
#each dep:
find_path(FLAC_INCLUDE_DIRS FLAC/all.h)  
find_library(FLAC_LIBRARIES FLAC)  
find_package_handle_standard_args(FLAC DEFAULT_MSG FLAC_INCLUDE_DIRS FLAC_LIBRARIES)
```
We can add them all together in the if(WIN32) block at the bottom like now, or add these last 3 as an else() in each of the BUILD_WITH_XXX blocks (that lets us get rid of the duplicate -DHAS_... definitions flag I added). Which way do you prefer? :)

The client/CMakeLists would be largely unchanged from badaix/master - only adding wasapi and windows libs.

Reply:
I would prefer the first option, to add them in the if(WIN32) block at the bottom

Reply:
sounds good!